### ISSUE TITLE
ruff 0.16.0 exceeds its own max line length when fixing use of format()

### ISSUE THREAD

**Issue by andy-maier** [astral-sh/ruff#27189]
### Summary

The new ruff version 0.16.0 exceeds its own max line length when fixing the use of format() to use f-strings.

Example: 

With max line length 80 set in the ruff config file:
```
line-length = 80
```

ruff, when used with `--fix`, makes this change:

```
     assert rc == exp_rc, \
-        "Unexpected exit code: got {}, expected {}\nstdout:\n{}\nstderr:\n{}". \
-        format(rc, exp_rc, stdout, stderr)
+        f"Unexpected exit code: got {rc}, expected {exp_rc}\nstdout:\n{stdout}\nstderr:\n{stderr}"
```

The resulting line is 98 characters long.

Strangely enough, ruff when checking the modified file does not complain about the line length of 98.

It is as if in both cases, it assumes a different max line length than the one specified in the `line-length` config parameter.

### Version

0.16.0

**Comment by Avasam** [comment]
This has also been a long time annoyance for me, but I don't think I ever opened an issue.

In early Ruff versions, it used to be that `.format` to f-string conversion would not even trigger (or as unsafe fix?) if it would make the line too long.

[printf-string-formatting (UP031)](https://docs.astral.sh/ruff/rules/printf-string-formatting/#printf-string-formatting-up031) I can enable unconditionally. But I find myself having a hard time with [f-string (UP032)](https://docs.astral.sh/ruff/rules/f-string/#f-string-up032) where readability suffers on lines with many replacements.

Would it be possible to configure [f-string (UP032)](https://docs.astral.sh/ruff/rules/f-string/#f-string-up032) to only trigger on short lines ? I could consider enabling it on pywin32's code if that was the case.

**Comment by ntBre** [comment]
I was going to say:

> I think our general assumption is actually that you will run a formatter after applying fixes, which should automatically resolve any line-length issues.

But that's not the case here. The formatter actually won't break up the long f-string.

Anyway, the tracking issue for this seems to be #8106. This was also discussed specifically for UP032 in https://github.com/astral-sh/ruff/issues/10259 and https://github.com/astral-sh/ruff/pull/10263.

Somewhat relatedly, if you do want to suppress the rule locally, our newly stabilized `ruff: ignore` comments are a good option since they can be used on their own line and not increase the line length further:

```py
def foo():
    # ruff: ignore[UP032]
     assert rc == exp_rc, \
        "Unexpected exit code: got {}, expected {}\nstdout:\n{}\nstderr:\n{}". \
        format(rc, exp_rc, stdout, stderr)
```
